File: app/src/main/java/closer/vlllage/com/closer/store/models/ReactionCount.java
package closer.vlllage.com.closer.store.models;

public class ReactionCount {
    public String reaction;
    public long count;

    @Override
    public boolean equals(Object other) {
        return other instanceof ReactionCount && reaction.equals(((ReactionCount) other).reaction) && count == ((ReactionCount) other).count;
    }
}

File: app/src/main/java/closer/vlllage/com/closer/store/models/ReactionCount.kt
package closer.vlllage.com.closer.store.models

class ReactionCount {
    var reaction: String? = null
    var count: Long = 0
    var preview: List<String> = listOf()

    override fun equals(other: Any?): Boolean {
        return other is ReactionCount &&
                reaction == other.reaction &&
                count == other.count &&
                preview == other.preview
    }

    override fun hashCode(): Int {
        var result = reaction?.hashCode() ?: 0
        result = 31 * result + count.hashCode()
        result = 31 * result + preview.hashCode()
        return result
    }
}

